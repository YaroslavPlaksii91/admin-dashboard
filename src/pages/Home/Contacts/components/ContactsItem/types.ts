import { MouseEventHandler } from 'react';

export type ContactType = {
  id: string;
  name: string;
  email: string;
  address: string;
  image: string;
  date: string;
};

export type ContactItemProps = {
  contact: ContactType;
  handleDelete: MouseEventHandler<HTMLDivElement>;
  handleEdit: MouseEventHandler<HTMLDivElement>;
};
