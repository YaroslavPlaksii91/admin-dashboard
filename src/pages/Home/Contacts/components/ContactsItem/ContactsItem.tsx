import { FC } from 'react';
import { Grid, Box, Typography } from '@mui/material';

import { formatDate } from '@services/date/formatDate';
import { SpeedDial } from '@pages/Home/components/SpeedDial/SpeedDial';

import { ContactItemProps } from './types';

export const ContactsItem: FC<ContactItemProps> = ({
  contact,
  handleDelete,
  handleEdit,
}) => {
  const date = formatDate(contact.date);

  return (
    <Grid
      container
      alignItems="center"
      sx={{
        padding: '24px 32px',
        borderBottom: '1px solid',
        borderBottomColor: 'borderColor',
        transition: 'background-color 250ms ease',
        '&:hover': {
          backgroundColor: 'rgba(55, 81, 255, 0.04)',
          cursor: 'pointer',
        },
      }}
    >
      <Grid item xs={4} sx={{ display: 'flex', alignItems: 'center' }}>
        <Box
          sx={{
            width: '44px',
            height: '44px',
            mr: 6,
            borderRadius: '50%',
            overflow: 'hidden',
          }}
        >
          <img src={contact.image} alt={contact.name} />
        </Box>

        <Typography variant="subtitle2" component="p">
          {contact.name}
        </Typography>
      </Grid>

      <Grid item xs={3}>
        <Typography variant="subtitle2" component="p">
          {contact.email}
        </Typography>
      </Grid>

      <Grid item xs={3} sx={{ pr: 5 }}>
        <Typography variant="subtitle2" component="p">
          {contact.address}
        </Typography>
      </Grid>

      <Grid
        item
        xs={2}
        sx={{
          display: 'flex',
          justifyContent: 'space-between',
          alignItems: 'center',
        }}
      >
        <Typography variant="subtitle2" component="p">
          {date.date}
        </Typography>

        <SpeedDial handleDelete={handleDelete} handleEdit={handleEdit} />
      </Grid>
    </Grid>
  );
};
