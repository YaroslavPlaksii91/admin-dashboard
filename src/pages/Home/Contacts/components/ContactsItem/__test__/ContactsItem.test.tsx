import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import { ContactsItem } from '../ContactsItem';

describe('ContactsItem component', () => {
  afterEach(cleanup);

  const handleDelete = jest.fn();
  const handleEdit = jest.fn();
  const contact = {
    id: '1',
    name: 'John Doe',
    email: 'johndoe@example.com',
    address: '123 Main St',
    date: '2023-10-10',
    image: '/path/to/image.jpg',
  };

  const props = {
    handleDelete,
    handleEdit,
    contact,
  };

  it('should be rendered', () => {
    const { asFragment } = render(<ContactsItem {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders contact information correctly', () => {
    render(<ContactsItem {...props} />);

    const nameElement = screen.getByText(contact.name);
    const emailElement = screen.getByText(contact.email);
    const addressElement = screen.getByText(contact.address);
    const dateElement = screen.getByText('Oct 10, 2023');

    expect(nameElement).toBeInTheDocument();
    expect(emailElement).toBeInTheDocument();
    expect(addressElement).toBeInTheDocument();
    expect(dateElement).toBeInTheDocument();
  });

  it('renders the contact image', () => {
    render(<ContactsItem {...props} />);

    const imageElement = screen.getByAltText(contact.name);

    expect(imageElement).toBeInTheDocument();
    expect(imageElement).toHaveAttribute('src', contact.image);
  });

  it('triggers handleDelete and handleEdit when SpeedDial buttons are clicked', () => {
    render(<ContactsItem {...props} />);

    const deleteButton = screen.getByLabelText('Delete');
    const editButton = screen.getByLabelText('Edit');

    fireEvent.click(deleteButton);
    fireEvent.click(editButton);

    expect(handleDelete).toHaveBeenCalled();
    expect(handleEdit).toHaveBeenCalled();
  });
});
