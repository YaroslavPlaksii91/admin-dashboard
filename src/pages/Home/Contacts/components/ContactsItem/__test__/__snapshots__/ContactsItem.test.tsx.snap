// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsItem component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container css-dkvi6p-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1dpyvte-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-2m3hzv"
      >
        <img
          alt="John Doe"
          src="/path/to/image.jpg"
        />
      </div>
      <p
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        John Doe
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        johndoe@example.com
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-8sbacc-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        123 Main St
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-n268fb-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        Oct 10, 2023
      </p>
      <div
        class="MuiSpeedDial-root MuiSpeedDial-directionUp css-5jgzwq-MuiSpeedDial-root"
        role="presentation"
      >
        <button
          aria-controls="SpeedDialmoreoptions-actions"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="SpeedDial more options"
          class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiSpeedDial-fab css-7dv1rb-MuiButtonBase-root-MuiFab-root-MuiSpeedDial-fab"
          style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          tabindex="0"
          type="button"
        >
          <svg
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
          >
            <use
              href="/sprite.svg#icon-more"
            />
            <title>
              More options
            </title>
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          aria-orientation="vertical"
          class="MuiSpeedDial-actions MuiSpeedDial-actionsClosed css-1s28u9o-MuiSpeedDial-actions"
          id="SpeedDialmoreoptions-actions"
          role="menu"
        >
          <button
            aria-label="Delete"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-tmh29y-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
            data-mui-internal-clone-element="true"
            role="menuitem"
            style="transition-delay: 60ms;"
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Edit"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-tmh29y-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
            data-mui-internal-clone-element="true"
            role="menuitem"
            style="transition-delay: 30ms;"
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EditIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
